setup/plugins/installFiles/StandaloneCivicrmFilesPath.civi-setup.php

```diff
@@ -0,0 +1,48 @@
+<?php
+/**
+ * @file
+ *
+ * Validate and create the civicrm.files folder.
+ */
+
+if (!defined('CIVI_SETUP')) {
+  exit("Installation plugins must only be loaded by the installer.\n");
+}
+
+\Civi\Setup::dispatcher()
+  ->addListener('civi.setup.checkRequirements', function (\Civi\Setup\Event\CheckRequirementsEvent $e) {
+    \Civi\Setup::log()->info(sprintf('[%s] Handle %s', basename(__FILE__), 'checkRequirements'));
+    $m = $e->getModel();
+
+    $civicrmFilesDirectory = $m->paths['civicrm.files']['path'] ?? '';
+
+    if (!$civicrmFilesDirectory) {
+      $e->addError('system', 'civicrmFilesPath', sprintf('The civicrmFilesPath is undefined.'));
+    }
+    else {
+      $e->addInfo('system', 'civicrmFilesPath', 'The civicrmFilesPath is defined.');
+    }
+
+    if ($civicrmFilesDirectory && !file_exists($civicrmFilesDirectory) && !\Civi\Setup\FileUtil::isCreateable($civicrmFilesDirectory)) {
+      $e->addError('system', 'civicrmFilesPathWritable', sprintf('The civicrm files dir "%s" does not exist and cannot be created. Ensure it exists or the parent folder is writable.', $civicrmFilesDirectory));
+    }
+    else {
+      $e->addInfo('system', 'civicrmFilesPathWritable', sprintf('The civicrm files dir "%s" can be created.', $civicrmFilesDirectory));
+    }
```

Comment:
good point! have added the conditions (I feel like it could be a useful notice for user to see that it's creating the directory where they want it/somewhere sensible)